### PR TITLE
KTOR-4524 Add logging for Routing

### PR DESCRIPTION
**Pull request by e5l**
Fix [KTOR-4524](https://youtrack.jetbrains.com/issue/KTOR-4524/Add-Debug-Logging-to-Routing)

### CODE REVIEW

**Comment by whyoleg** [comment]
Just wanted to mention that on current moment we don't have possibility to filter logs on nix server. On JVM we have slf4j and so we can have debug/trace configuration overall and based on class/tag name.
So adding a lot of logs, even TRACE, will significantly reduce usability of nix server.

**Comment by e5l** [comment]
@whyoleg we're going to add logs in 2.2.0 along with the `log level` option for native.

**Comment by whyoleg** [comment]
@e5l Is there any info on how it will work?

**Comment by e5l** [comment]
It will be possible to set with ENV variable or configuration file

File: ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/routing/RoutingResolveContext.kt [hunk 1]
```diff
@@ -170,6 +175,7 @@ public class RoutingResolveContext(
         val finalResolve = resolveResult
 
         if (finalResolve.isEmpty()) {
+            LOGGER.trace("No route matched for ${call.request.uri}")
```
Comment (rsinukov) [inline ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/routing/RoutingResolveContext.kt:178]: why do we need both logging and tracing in routing? can they be merged?

**Review by rsinukov — CHANGES_REQUESTED** [review]

File: ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/routing/RoutingResolveContext.kt (hunk 1 above)
Comment (rsinukov) [reply ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/routing/RoutingResolveContext.kt:178]: Tracing has more information and is better structured. Maybe we can enable it automatically, if log level is trace?

Comment (e5l) [reply ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/routing/RoutingResolveContext.kt]: Done!

**Pull request merged**